File: platform/Leetcode/src/findMaxProfit.java
import java.util.Scanner;

public class findMaxProfit{
    public static void main(String[] args) {
        System.out.print("Enter a list of integers (prices), separated by commas: ");
        Scanner sc = new Scanner(System.in);
        String input = sc.nextLine();
        
        String[] numbers = input.split(",");
        
        int[] numberInput = new int[numbers.length];
        for (int i=0; i<numbers.length; i++) {
            numberInput[i] = Integer.parseInt(numbers[i]);
        }
          
        System.out.println("The best day to sell is : " + findMaxProfit(numberInput))  ;      
        
    }
    
    public static int findMaxProfit(int[] prices) {
        if (prices.length == 0) {
            return 0;
        }
        int maxProfit = 0;
        int basePrice = prices[0];

        for (int i=1; i<prices.length; i++) {
            int profit = 0;
            if ((prices[i] - basePrice) > 0) {
                profit = (prices[i] - basePrice);
            } else {
                profit = maxProfit;
            }

            if (prices[i] < basePrice) {
                basePrice = prices[i];
            }

            if (profit > maxProfit) {
                maxProfit = profit;
            }
        }
        return maxProfit;
    }
}

File: platform/Leetcode/src/find_max_profit.py
def find_max_profit(prices):
    if len(prices) == 0:
        return 0
    
    max_profit = 0
    base_price = prices[0]
    
    for i in range(1, len(prices)):        
            
        if (prices[i] - base_price) > 0:
            profit = (prices[i] - base_price) 
        else:
            profit = max_profit        
        
        if prices[i] < base_price:
            base_price = prices[i]        
        
        if profit > max_profit:
            max_profit = profit

    return max_profit

def convert_to_list(string):
    return [int(each.strip()) for each in string.split(',') if each != '']

def main():
    user_input = input("Enter a list of integers, separated by commas: ")
    numbers = convert_to_list(user_input)
    output = find_max_profit(numbers)
    print("The group and the people related ids are: ", output)
    
if __name__=='__main__':
    main()
    
